### PR TITLE
Hardcoded

### PR DESCRIPTION
**Pull request by santhiya-mj**
- Replaced all hardcoded ports with environment variables across multiple files.

- Created a new .env file to store environment variables.

- Loaded the .env file in: integrationtests.yml, kill_server.sh, run_server.sh

- I added a new dependency to requirements.txt to support loading environment variables for integration tests.

- Updated files to replace hardcoded ports with environment variables: database_manager.py, minio_manager.py, main.py, weaviate_manager.py, docker-compose.yml

- Added .env file loading to all test scripts: test_addcustomer.py, test_chat.py, test_deletechat.py, test_downloadfile.py, test_getallchats.py, test_listfile.py, test_ollamaserver.py, test_uploadfile.py

### CODE REVIEW

File: src/backend/.env [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Chat Service Configuration
+DB_USER=root
```
Comment (SaravanakumarR2018) [inline src/backend/.env:2]: Rename DB_USER=root as MYSQL_ROOT_USER=root and replace it everywhere
REMOVE DB_PASSWORD and use existing MYSQL_ROOT_PASSWORD wherever DB_PASSWORD is used
Remove DB_HOST and make it MYSQL_HOST=mysql_db and replace all its usages

File: src/backend/docker-compose.yml [hunk 2]
```diff
@@ -119,4 +121,3 @@ volumes:
 # Define the absolute path for PROJECT_ROOT
 x-environment:
   PROJECT_ROOT: ${PROJECT_ROOT}
-
```
Comment (SaravanakumarR2018) [inline src/backend/docker-compose.yml:122]: Don't make this change

File: src/backend/main/main.py [hunk 3]
```diff
@@ -49,4 +49,7 @@ async def check_server_status(request: Request):
 # Run this file to start the server
 if __name__ == "__main__":
     import uvicorn
-    uvicorn.run(main_app, host="0.0.0.0", port=8000, reload=True)
+    import os
+
+    port = int(os.environ.get("CHAT_SERVICE_PORT"))
```
Comment (SaravanakumarR2018) [inline src/backend/main/main.py:54]: rename as chat_service_port

File: test/IntegrationTests/test_addcustomer.py [hunk 4]
```diff
@@ -1,13 +1,19 @@
 import unittest
 import requests
 import logging
+import os
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+dotenv_path = os.path.join(os.path.dirname(__file__), '..', '..', '..', 'primitivechat', 'src', 'backend', '.env')
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_addcustomer.py:8]: Make it dotenv_path = os.path.join(os.path.dirname(__file__), '..', '..', 'src', 'backend', '.env') everywhere

File: .github/workflows/integrationtests.yml [hunk 5]
```diff
@@ -21,6 +21,49 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y docker-compose
 
+      - name: Load .env and verify
```
[the remark this answers is not in the document]
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml:25]: I completely removed that and also replaced this step

File: .github/workflows/integrationtests.yml [hunk 6]
```diff
@@ -58,12 +101,12 @@ jobs:
             fi
           done
           echo "Error: Server did not start within 2 minutes."
-          exit 1
```
[the remark this answers is not in the document]
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml]: okay

File: build/kill_server.sh [hunk 7]
```diff
@@ -13,6 +13,15 @@ echo "Project root: $PROJECT_ROOT"
 export PROJECT_ROOT
 echo "PROJECT_ROOT is set to: $PROJECT_ROOT"
 
+# Load environment variables from .env file
```
[the remark this answers is not in the document]
Comment (santhiya-mj) [reply build/kill_server.sh:16]: I completely removed that and also replaced this step

File: build/run_server.sh [hunk 8]
```diff
@@ -11,6 +11,15 @@ PROJECT_ROOT="$SCRIPT_DIR/.."  # Use the local directory structure
 export PROJECT_ROOT
 echo "PROJECT_ROOT is set to: $PROJECT_ROOT"
 
+# Load environment variables from .env file
```
[the remark this answers is not in the document]
Comment (santhiya-mj) [reply build/run_server.sh:14]: I completely removed that and also replaced this step

File: src/backend/.env (hunk 1 above)
Comment (santhiya-mj) [reply src/backend/.env]: I've renamed DB_USER to MYSQL_ROOT_USER, replaced DB_PASSWORD with MYSQL_ROOT_PASSWORD, and updated DB_HOST to MYSQL_HOST.

File: src/backend/docker-compose.yml (hunk 2 above)
Comment (santhiya-mj) [reply src/backend/docker-compose.yml:122]: okay

File: src/backend/main/main.py (hunk 3 above)
Comment (santhiya-mj) [reply src/backend/main/main.py]: I renamed this to chat_service_port in the code.

File: test/IntegrationTests/test_addcustomer.py (hunk 4 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_addcustomer.py]: I have replaced this line everywhere

File: src/backend/main/main.py [hunk 9]
```diff
@@ -49,4 +49,7 @@ async def check_server_status(request: Request):
 # Run this file to start the server
 if __name__ == "__main__":
     import uvicorn
-    uvicorn.run(main_app, host="0.0.0.0", port=8000, reload=True)
+    import os
+
+    CHAT_SERVICE_PORT = int(os.environ.get("CHAT_SERVICE_PORT"))
```
Comment (SaravanakumarR2018) [inline src/backend/main/main.py:54]: Make it chat_service_port = int(os

Make it small letters

File: test/IntegrationTests/test_addcustomer.py [hunk 10]
```diff
@@ -1,13 +1,19 @@
 import unittest
 import requests
 import logging
+import os
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+dotenv_path = os.path.join(os.path.dirname(__file__), '..', '..', 'src', 'backend', '.env')
+load_dotenv(dotenv_path)
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_addcustomer.py:9]: Remove these lines, the code will work without these lines

Make sure you remove these lines in all the files

**Comment by SaravanakumarR2018** [comment]
Do a search across the entire files to check that there is no port or ip hard coding

File: src/backend/main/main.py (hunk 9 above)
Comment (santhiya-mj) [reply src/backend/main/main.py]: I changed it to lowercase chat_service_port

File: test/IntegrationTests/test_addcustomer.py (hunk 10 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_addcustomer.py]: I removed those three lines from all files

**Review by SaravanakumarR2018 — APPROVED** [review]
I have approved this pull request 
You can merge it

Change this line in a separate pull request and send a separate pull request

[src/backend/docker-compose.yml](https://github.com/SaravanakumarR2018/primitivechat/pull/81/files#diff-8038805d27c024521d365351c85b2443523944bff12b8938bd20354df23d0308)
Viewed
Original file line number	Diff line number	Diff line change
@@ -5,20 +5,20 @@ services:
    container_name: chat_service
    working_dir: /app
    environment:
      DB_USER: ${DB_USER:-root}
      DB_PASSWORD: ${DB_PASSWORD:-admin123}
      DB_HOST: ${DB_HOST:-mysql_db}
      MYSQL_ROOT_USER: ${MYSQL_ROOT_USER}
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      **MYSQL_HOST: mysql_db. => MYSQL_HOST: ${MYSQL_HOST}**

**Pull request merged**